components/wifi/esp8266-driver/ESP8266Interface.cpp

```diff
@@ -246,7 +246,7 @@ int ESP8266Interface::disconnect()
 {
     _initialized = false;
 
-    if (_conn_stat == NSAPI_STATUS_DISCONNECTED)
+    if (_conn_stat == NSAPI_STATUS_DISCONNECTED || !get_ip_address())
```

Comment:
I understand it happenned that get_ip_address() was false, even though _conn_stat was not DISCONNECTED.
But is the reverse situation possible? Can get_ip_address() return true when _conn_stat is DISCONNECTED?
Perhaps there is no point checking the _conn_stat here?

Reply:
If the modem isn't powered on this prevent's the situation that AT parser's recv function sits there for the length of ESP8266_RECV_TIMEOUT waiting for answer to the request.